### PR TITLE
Making file_server reusable

### PR DESCRIPTION
**Pull request by avinashdvv**
@kevinkassimo

### CODE REVIEW

**Comment by hayd** [comment]
On std convention: it seems strange that `net/file_server.ts` would be an executable, I wonder if this should be in `file_server/main.ts` or ... something else `bin/file_server.ts`? I realise this was briefly discussed in the earlier PR, but this was before #66

File: net/file_server.ts [hunk 1]
```diff
@@ -58,202 +24,9 @@ if (target) {
...
-  return `
-  <tr><td class="mode">${modeToString(
-    isDir,
-    mode
-  )}</td><td>${sizeStr}</td><td><a href="${path}">${name}${
-    isDir ? "/" : ""
-  }</a></td>
-  </tr>
-  `;
+if (fileServer) {
```
Comment (bartlomieju) [inline net/file_server.ts]: This condition is not needed, `fileServer` is an import and will always be defined.

Comment (avinashdvv) [reply net/file_server.ts]: how can i achieve this kind of feature in deno
```
def fileServer(...):
   ...

if __name__ == '__main__':
    fileServer(...)
```

Comment (bartlomieju) [reply net/file_server.ts]: You can't, and it's actually not needed, the code you wrote is what we want 👍

**Comment by kevinkassimo** [comment]
@avinashdvv Can you fix the failing tests?

**Comment by hayd** [comment]
To push one more time here: I think this belongs in examples. So the exported `fileServer` function is in `http/file_server.ts` and the actual run-able server is `examples/file_server.ts`.

(Personally I am not of fan of files > 400LOC if can be easily avoided.)

**Comment by ry** [comment]
I don't like having all this html stuff in `http.ts` - I think it ought to be as lean as possible. I'm all for making file_server reusable tho.

I'm closing this as this PR is dated - let's discuss further if you want to still land this change - or just open a new PR.

**Pull request closed without merging**